pkg/test/framework/components/istioctl/istioctl.go

```diff
@@ -0,0 +1,90 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package istioctl
+
+import (
+	"bytes"
+	"testing"
+
+	"istio.io/istio/istioctl/cmd"
+
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+type Instance interface {
+	// Invoke invokes an istioctl command and returns the output and exception.
+	// Cobra commands don't make it easy to separate stdout and stderr and the string parameter
+	// will receive both.
+	Invoke(args []string) (string, error)
+}
+
+// Structured config for the istioctl component
+type Config struct {
+	// currently nothing, we might add stuff like OS env settings later
+}
+
+// The test code will be linked with istioctl, and never runs a seperate istioctl binary
+type eitherComponent struct {
+	id     resource.ID
+	ctx    resource.Context
+	config Config
+}
+
+// New returns a new instance of "istioctl".
+func New(ctx resource.Context, cfg Config) (i Instance, err error) {
+	err = resource.UnsupportedEnvironment(ctx.Environment())
+	ctx.Environment().Case(environment.Native, func() {
```

Comment:
Do you want to make istioctl work in native environment? It means there won't be a cluster, and all components will be running locally.

It maybe meaningful for istioctl, I just want to call it out and let you make an informed decision.

Reply:
Some Istio commands, like `istioctl validate`, do not require any Pilot or cluster.  I left the native case in because these commands should work.

Other commands like `kube-inject` don't work without a cluster.  I considered leaving out the native case for this reason, but decided that the TestMain's themselves should control that.